.github/workflows/build-sample-apps.yml

```diff
@@ -83,11 +88,19 @@ jobs:
         ruby-version: '3.0'
         bundler-cache: true # cache tools to make builds faster in future 
 
-    - name: Setup build environment with Customer.io workspace credentials 
+    - name: Setup Customer.io workspace credentials for Java 
+      if: ${{ matrix.cio-workspace == 'java' }}
       run: |
         touch "samples/local.properties"
-        echo "siteId=${{ secrets.CIO_SITE_ID }}" >> "samples/local.properties"
-        echo "apiKey=${{ secrets.CIO_API_KEY }}" >> "samples/local.properties"          
+        echo "siteId=${{ secrets.CUSTOMERIO_JAVA_WORKSPACE_SITE_ID }}" >> "samples/local.properties"
+        echo "apiKey=${{ secrets.CUSTOMERIO_JAVA_WORKSPACE_API_KEY }}" >> "samples/local.properties" 
+
+    - name: Setup Customer.io workspace credentials for Kotlin
+      if: ${{ matrix.cio-workspace == 'kotlin' }}
+      run: |
+        touch "samples/local.properties"
+        echo "siteId=${{ secrets.CUSTOMERIO_KOTLIN_WORKSPACE_SITE_ID }}" >> "samples/local.properties"
+        echo "apiKey=${{ secrets.CUSTOMERIO_KOTLIN_WORKSPACE_API_KEY }}" >> "samples/local.properties"          
```

Comment:
can we add `samples/local.properties` into an `env` variable?
It would make it much easier to change in all places and the chances of messing up the spelling on one of the occurrences gets lower as well.

Reply:
```
env:
      LOCAL_PROP_FILE: "samples/local.properties"
```

Reply:
We did that strategy in RH and I strongly recommend we don't do that. That strategy required versioning the secret file and gets complex quickly. Also, there is an added risk of credentials getting leaked when you do not save the credential values as secrets. Another reason to do this approach instead. 

However, I do not like the copy/paste nature of the workflow as-is. 

I [found a way](https://github.com/customerio/customerio-android/pull/236/commits/c5a77546f69174ff12ae3b62d89ba063d2c9eaf4) to dynamically reference secrets which allows the workflow to be dynamic with it's generating of the local.properties file.